src/clusterfuzz/fuzz/__main__.py

```diff
@@ -47,28 +46,28 @@ def main():
       help='Max time in seconds to run.',
       type=int,
       default=3600)
+  parser.add_argument('engine_args', nargs='*')
   args = parser.parse_args()
 
-  # TODO(ochang): Find a cleaner way to propagate this.
-  if args.sanitizer == 'address':
-    os.environ['JOB_NAME'] = 'libfuzzer_asan'
-  elif args.sanitizer == 'memory':
-    os.environ['JOB_NAME'] = 'libfuzzer_msan'
-  elif args.sanitizer == 'undefined':
-    os.environ['JOB_NAME'] = 'libfuzzer_ubsan'
+  with Environment(args.engine, args.sanitizer, args.target, interactive=True):
+    engine_impl = clusterfuzz.fuzz.get_engine(args.engine)
+    options = engine_impl.prepare(args.corpus, args.target,
+                                  os.path.dirname(args.target))
+    if args.engine_args:
+      options.arguments.extend(args.engine_args)
 
-  os.environ['BUILD_DIR'] = os.path.dirname(args.target)
+    if args.engine == 'libFuzzer':
+      options.merge_back_new_testcases = False
+      options.analyze_dictionary = False
 
-  engine_impl = clusterfuzz.fuzz.get_engine(args.engine)
-  options = engine_impl.prepare(args.corpus, args.target,
-                                os.path.dirname(args.target))
-  if args.engine == 'libFuzzer':
-    options.merge_back_new_testcases = False
-    options.analyze_dictionary = False
+    result = engine_impl.fuzz(args.target, options, args.output,
+                              args.max_duration)
 
-  result = engine_impl.fuzz(args.target, options, args.output,
-                            args.max_duration)
   print('Command: ', ' '.join([shlex.quote(part) for part in result.command]))
+  if result.crashes:
+    sys.exit(1)
```

Comment:
I'd prefer we returned these values and that `sys.exit(main())` was called on line 74

Reply:
That's a much bigger change, and not possible today without a lot of refactoring.